aws-ts-static-website/index.ts

```diff
@@ -79,6 +78,38 @@ const logsBucket = new aws.s3.Bucket("requestLogs",
 
 const tenMinutes = 60 * 10;
 
+let certificateArn: pulumi.Input<string> = config.certificateArn!;
+if (config.certificateArn === undefined) {
+    // Per AWS, ACM certificate must be in the us-east-1 region.
+    const eastRegion = new aws.Provider("east", {
+        region: "us-east-1",
+    });
+    const certificate = new aws.acm.Certificate("certificate", {
+        domainName: config.targetDomain,
+        validationMethod: "DNS",
+    }, { provider: eastRegion });
+
+    const domainParts = getDomainAndSubdomain(config.targetDomain);
+    const hostedZoneId = aws.route53.getZone({ name: domainParts.parentDomain }).then(zone => zone.id);
+
+    const certificateValidationDomain = new aws.route53.Record(
+        `${config.targetDomain}-validation`,
+        {
+            name: certificate.domainValidationOptions.apply(d => d[0].resourceRecordName),
+            zoneId: hostedZoneId,
+            type: certificate.domainValidationOptions.apply(d => d[0].resourceRecordType),
+            records: [certificate.domainValidationOptions.apply(d => d[0].resourceRecordValue)],
+            ttl: tenMinutes,
+        });
+
+    const certificateValidation = new aws.acm.CertificateValidation("certificateValidation", {
+        certificateArn: certificate.arn,
+        validationRecordFqdns: [certificateValidationDomain.fqdn],
+    }, { provider: eastRegion });
+
+    certificateArn = certificateValidation.certificateArn;
+}
+
 // distributionArgs configures the CloudFront distribution. Relevant documentation:
 // https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/distribution-web-values-specify.html
 // https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html
```

Comment:
Would you mind adding a comment on top of the `aliases` attribute here too, just to make the next time this example is built upon:

```diff
 const distributionArgs: aws.cloudfront.DistributionArgs = {
     enabled: true,	     enabled: true,
+    // Alternate aliases the CloudFront distribution can be reached at, in addition to https://xxxx.cloudfront.net.
+    // Required if you want to access the distribution via config.targetDomain as well.  
     aliases: [ config.targetDomain ],
```

Reply:
Added!